### PR TITLE
CORTX-31410 : Add overall regression status in nightly deployment

### PR DESCRIPTION
**Pull request by gauravchaudhari02**
Signed-off-by: Gaurav Chaudhari <gaurav.chaudhari@seagate.com>

# Problem Statement
- Multiple regression status inputs are coming from QA job and hence it is difficult to provide overall regression status

# Design
-  Take overall regression status from QA job and provide it to email template

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [ ] Jenkins pipelines used for testing

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/internal-ci/generic/nightly/nightly-deploy.groovy [hunk 1]
```diff
@@ -152,7 +152,8 @@ pipeline {
                         env.Sanity_Failed = qaSanity.buildVariables.Sanity_Failed
                         env.Sanity_status = qaSanity.buildVariables.Sanity_Failed.toString() == 'true' ? 'Failed' : qaSanity.buildVariables.Sanity_Failed.toString() == 'false' ? 'Passed' : 'Skipped'
                         env.Regression_Failed = qaSanity.buildVariables.Regression_Failed
-                        env.Regression_status = qaSanity.buildVariables.Regression_Failed.toString() == 'true' ? 'Failed' : qaSanity.buildVariables.Regression_Failed.toString() == 'false' ? 'Passed' : 'Skipped'
+                        env.Regression_overall_failed = = qaSanity.buildVariables.Regression_overall_failed
```
Comment (akshaym99) [inline jenkins/internal-ci/generic/nightly/nightly-deploy.groovy:155]: This assignment is not require. As on L156 assigning it.

Comment (gauravchaudhari02) [reply jenkins/internal-ci/generic/nightly/nightly-deploy.groovy]: Done

**Review by shailesh-vaidya — COMMENTED** [review]
Looks good but could you please validate with test run

File: jenkins/internal-ci/generic/nightly/nightly-deploy.groovy (hunk 1 above)
Comment (akshaym99) [reply jenkins/internal-ci/generic/nightly/nightly-deploy.groovy]: Looks good to me

**Review by akshaym99 — APPROVED** [review]

**Comment by gauravchaudhari02** [comment]
> Looks good but could you please validate with test run

@shailesh-vaidya, Tested and sent mail notification to you

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**